### DISCUSSION TITLE
Controller's defaultValue for radio inputs / checkboxes

### DISCUSSION THREAD

**Discussion by vrazn** [react-hook-form/react-hook-form#2482] [Q&A]
Hello!

Since the new version, the `Controller` started to throw warnings when `defaultValues` are not defined in both `useForm()` and the `Controller` itself.

Let's say I have a user object with a structure like this:
```
{ 
  personalInformation: {
    employment?: 'employed' | 'self-employed' | 'unemployed',
    someOtherField?: any,
  }
}
```
And a component like this:
![image](https://user-images.githubusercontent.com/8733843/89384000-0295df80-d730-11ea-9efd-0d5bdeb85179.png)

In our current implementation, the component's initial value is `undefined`. When the user selects something it becomes one of `employed`, `self-employed`, or `unemployed`.

What is the best way to work around it now? Setting `defaultValues` to null or empty strings just to skip the warning?
Predefining a value to one of those three is not an option, we don't want the component to have a pre-selected value.
And why exactly does the `Controller` always need a `defaultValue`?

**Comment by bluebill1049** [comment]
we always need a defaultvalue, because it's uncontrolled, hence you wouldn't see re-render while interactive with the inputs.
